pkg/composer/exec.go

```diff
@@ -54,9 +58,29 @@ func (c *Composer) Exec(ctx context.Context, eo ExecOptions) error {
 		return fmt.Errorf("index (%d) out of range: only %d running instances from service %s",
 			eo.Index, len(containers), eo.ServiceName)
 	}
-	container := containers[eo.Index-1]
+	if len(containers) == 1 {
+		return c.exec(ctx, containers[0], eo)
+	}
+	// The sort of containers is not stable
+	// so we compare the index by its name
+	for _, container := range containers {
+		info, err := container.Info(ctx, containerd.WithoutRefreshedMetadata)
+		if err != nil {
+			if errdefs.IsNotFound(err) {
+				continue
+			}
+			return fmt.Errorf("fail to get container info for service: %s: %s", eo.ServiceName, err)
+		}
+		name := info.Labels[labels.Name]
+		segs := strings.Split(name, "-")
```

Comment:
nit: uses the const instead of plain string `"-"`

https://github.com/containerd/nerdctl/blob/0fbf4967b3bea2a1f8bc7212aa3390abae091bf8/pkg/composer/serviceparser/serviceparser.go#L52

Reply:
make sense